src/tests/Common/CoreCLRTestLibrary/Vectors.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace TestLibrary
+{
+    public static class Vectors
+    {
+        public static void VectorToArray<T>(ref T[] to, Vector<T> from)
+        {
+            long tsize = Unsafe.SizeOf<T>();
+            long vsize = Unsafe.SizeOf<Vector<T>>();
+            if (to.Length * tsize != vsize)
+            {
+                throw new ArgumentException("sizeof(to) != sizeof(from)");
+            }
+            Unsafe.WriteUnaligned(ref Unsafe.As<T, byte>(ref to[0]), from);
```

Comment:
I think we have to pin the array if we're going to use `Unsafe.*`, right?

Since we're here, I'd prefer we rewrite this without all the `Unsafe.*` usage. Something like:

```suggestion
            Span<bytes> dst = MemoryMarshal.AsBytes(to.AsSpan());
            MemoryMarshal.Write(dst, ref from);
```